medusa/server/api/v2/history.py

```diff
@@ -0,0 +1,100 @@
+# coding=utf-8
+"""Request handler for alias (scene exceptions)."""
+from __future__ import unicode_literals
+
+from medusa import db
+
+from medusa.server.api.v2.base import BaseRequestHandler
+from medusa.providers.generic_provider import GenericProvider
+from medusa.providers import get_provider_class
+from medusa.tv.series import SeriesIdentifier
+from os.path import basename
+from medusa.common import statusStrings
+
+
+class HistoryHandler(BaseRequestHandler):
+    """History request handler."""
+
+    #: resource name
+    name = 'history'
+    #: identifier
+    identifier = ('series_slug', r'\w+')
+    #: path param
+    path_param = ('path_param', r'\w+')
+    #: allowed HTTP methods
+    allowed_methods = ('GET', 'POST', 'PUT', 'DELETE')
+
+    def get(self, series_slug, path_param):
+        """
+        Get history records.
+
+        History records can be specified using a show slug.
+        """
+        sql_base = '''
+            SELECT rowid, date, action, quality,
+                   provider, version, resource, size,
+                   indexer_id, showid, season, episode
+            FROM history
+        '''
+        params = []
+
+        arg_page = self._get_page()
+        arg_limit = self._get_limit(default=50)
+
+        if series_slug is not None:
+            series_identifier = SeriesIdentifier.from_slug(series_slug)
+            if not series_identifier:
+                return self._bad_request('Invalid series')
+
+            sql_base += ' WHERE indexer_id = ? AND showid = ?'
+            params += [series_identifier.indexer.id, series_identifier.id]
+
+        sql_base += ' ORDER BY date DESC'
+        results = db.DBConnection().select(sql_base, params)
+
+        def data_generator():
+            """Read log lines based on the specified criteria."""
+            start = arg_limit * (arg_page - 1) + 1
```

Comment:
If you remove the +1 here you can remove the two -1 below

Reply:
Hmm, not sure. For ex. Page 1, with a Limit of 50. Will result in 1, page 2 will result in 51

Reply:
There was a reason I did it like this. Can't remember. I'll try it out

Reply:
There was a reason I did it like this. Can't remember. I'll try it out

Reply:
I'll try it out